### ISSUE TITLE
bug(linter): explicit-length-check inside ternary

### ISSUE THREAD

**Issue by johnrazeur** [oxc-project/oxc#2156]
I've got a false positive with `explicit-length-check` when size is inside a ternary.

```ts
  const totalCount = tests.reduce(
    (count, test) => count + (test.enabled ? test.maxSize : test.size),
    0,
  )
```

```
  ⚠ eslint-plugin-unicorn(explicit-length-check): Use `.size > 0` when checking size is not zero.
     ╭─[src/...]
 110 │   const totalCount = tests.reduce(
 111 │     (count, test) => count + (test.enabled ? test.maxSize : test.size),
     ·                                                               
 112 │     0,
     ╰────
```
